### PR TITLE
doc: Update README and remove graphql-js references

### PR DESCRIPTION
**Pull request by addityasingh**
Update README and remove graphql-js references

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -101,17 +101,16 @@ console.log(executionResult);
...
-- `schema` {GraphQLSchema} - `graphql-js` schema object
-- `document` {DocumentNode} - document query AST ,can be obtained by `parse` from `graphql-js`
-- `operationName` {string} - optional operation name in case the document contains multiple queries/operations.
-- `compilerOptions` {Object} - Configurable options on the agent pool
+- `schema` {GraphQLSchema} - `graphql` schema object
+- `document` {DocumentNode} - document query AST ,can be obtained by `parse` from `graphql`
+- `operationName` {string} - optional operation name in case the document contains multiple [operations](http://spec.graphql.org/draft/#sec-Language.Operations)(queries/mutations/subscription).
+- `compilerOptions` {Object} - Configurable options for the compiler
 
   - `disableLeafSerialization` {boolean, default: false} - disables leaf node serializers. The serializers validate the content of the field
```
Comment (addityasingh) [inline README.md]: Explanation for this option is not clear. How is disabling leaf serialization related to validation?

File: README.md [hunk 2]
```diff
@@ -27,15 +27,15 @@ graphql-jit x 1,317 ops/sec ±2.38% (213 runs sampled)
...
-the only missing feature is support for the @skip and @include directives.
+the only missing feature is support for the `@skip` and `@include` directives.
 
-#### Differences to graphql-js
+#### Differences to `graphql-js`
 
-In order to achieve better performance, the compiler introduces some limitations.
-The main one is that all computed properties must have a resolver and only these can return a Promise.
+In order to achieve better performance, the `graphql-jit` compiler introduces some limitations.
+The primary limitation is that all computed properties must have a resolver and these can only return a Promise or a function.
```
Comment (boopathi) [inline README.md]:
```suggestion
The primary limitation is that all computed properties must have a resolver and only these can return a `Promise`.
```

File: README.md (hunk 1 above)
Comment (ruiaraujo) [reply README.md]: runtime validation of the output types is handled during the serialization of the field.

Comment (addityasingh) [inline README.md]:
```suggestion
  - `disableLeafSerialization` {boolean, default: false} - disables leaf node serializers. The serializers validate the content of the field at runtime
```

**Comment by ruiaraujo** [comment]
:+1:

**Comment by addityasingh** [comment]
👍

**Pull request merged**